### DISCUSSION TITLE
web assets not found, please make sure they have been built

### DISCUSSION THREAD

**Discussion by hard-game** [cncf/landscape2#757] [Q&A]
Under what circumstances does this situation occur?

use landscape2，in some machine，it work，but in other，it notwork

**Comment by hard-game** [comment]
i build a landscape2 from source code by cargo build
if i need more args to use cargo build?

**Comment by tegioz** [comment]
Hi @hard-game

No, you shouldn't need to pass more arguments to cargo build. Web assets are built by the `cli` crate [build script](https://github.com/cncf/landscape2/blob/main/crates/cli/build.rs) automatically when you run `cargo build`. You'll need `yarn` and `wasm-pack` installed on the machine. If something goes wrong during that process, you should see some information about the error in the output.

**Reply by hard-game** [reply]
The current situation is that I downloaded the source code for landscape2 and ran cargo build in the local /landscape2 directory. After that, I uploaded the generated executable file target/debug/landscape2 to the remote server to compile landscape2. In this case, due to the missing files under /ui/webapp/dist/, this error occurred. However, the v0.11.0 release that I downloaded from GitHub does not have this issue. What should I do to make the packaged files work like the ones you provided?

**Reply by hard-game** [reply]
i try to use #cargo build --release to fix it

**Reply by tegioz** [reply]
During the landscape2 build process (when you run `cargo build`), the built static assets (which are built by the build script I shared before) are embedded into the final binary. What you are doing is correct, and you don't need to upload anything else to the remote server, just the final landscape2 binary.

For some reason, it looks like the web assets are not being built properly. Please take a look at the full output of the `cargo build` process to check if you see something suspicious that can help debugging the problem. You can start from a fresh clone if you'd like.

**Reply by hard-game** [reply]
I use cargo build --release fix it ,now it can build success
in build.rs,i see profile,is this affect?

**Reply by tegioz** [reply]
Hi @hard-game

The build process should work fine on `debug` and `release` modes, no need to use release until that's actually needed (when you are done developing and ready to ship a final version). It'd be good to see the errors you're getting to see what's really going on in those cases the build isn't working.

The profile you see in the `build.rs` script is used to build the `wasm` crate using the same profile passed to the `cli` crate. Nothing to worry about.

**Comment by hard-game** [comment]
try use cargo build --release
